packages/core/src/NavigationFocusedContext.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { NavigationHelpers, ParamListBase } from './types';
+
+export type FocusedNavigationCallback = (
+  navigation: NavigationHelpers<ParamListBase>
+) => any;
+
+export type FocusedNavigationListener = (
+  callback: FocusedNavigationCallback
+) => { handled: boolean; result: any };
+
+const MISSING_CONTEXT_ERROR =
+  "We couldn't find a navigation focused context. Have you wrapped your app with 'NavigationContainer'?";
+
+/**
+ * Context which holds the required helpers needed to access focused navigator.
+ */
+
+const NavigationFocusedContext = React.createContext<{
+  addNavigationListener: (listener: FocusedNavigationListener) => void;
+}>({
+  get addNavigationListener(): any {
+    throw new Error(MISSING_CONTEXT_ERROR);
+  },
+});
+
+export default NavigationFocusedContext;
+
+/**
+ * Hook which lets child navigators add action listeners.
+ */
+export function useFocusedListeners() {
```

Comment:
Let's move this hook to separate file like other hooks. We could move the listener logic you added in `useNavigationHelpers` into this hook because they are related logic.